scripts/install-latest-ci.sh

```diff
@@ -1,101 +1,148 @@
     API="repos/$OWNER/$REPO/actions/artifacts/$ARTIFACT_ID/zip"
-    gh api "$API" > "$TMPDIR/$FILENAME"
+    gh api "$API" > "$filename"
 }
 
-ARCH=x86_64
-if [ "$(uname -m)" = "arm64" ]; then
-    ARCH=aarch64
-fi
-download_artifact "Rancher Desktop-mac.$ARCH.zip"
-
-# Artifacts are zipped, so extract inner ZIP file from outer wrapper.
-# The outer ZIP has a predictable name like "Rancher Desktop-mac.x86_64.zip"
-# but the inner one has a version string: "Rancher Desktop-1.7.0-1061-g91ab3831-mac.zip"
-# Extract filename from:
-# ---------------------------------------------------------------------------
-# $ unzip -l Rancher\ Desktop-mac.x86_64.zip
-# Archive:  Rancher Desktop-mac.x86_64.zip
-#   Length      Date    Time    Name
-# ---------  ---------- -----   ----
-# 617701842  04-18-2023 23:56   Rancher Desktop-1.7.0-1061-g91ab3831-mac.zip
-# ---------                     -------
-# 617701842                     1 file
-# ---------------------------------------------------------------------------
-ZIP=$(unzip -l "$TMPDIR/$FILENAME" | perl -ne 'print if s/^\d.*Rancher/Rancher/')
-if [ -z "$ZIP" ]; then
-    echo "Cannot find inner archive in $TMPDIR/$FILENAME"
-    exit 1
-fi
+wslpath_from_win32_env() {
+    # The cmd.exe _sometimes_ returns an empty string when invoked in a subshell
+    # wslpath "$(cmd.exe /c "echo %$1%" 2>/dev/null)" | tr -d "\r"
+    # Let's see if powershell.exe avoids this issue
+    wslpath "$(powershell.exe -Command "Write-Output \${Env:$1}")" | tr -d "\r"
+}
 
-unzip -o "$TMPDIR/$FILENAME" "$ZIP" -d "$TMPDIR"
+install_application() {
+    local archive
+    if [[ "$(uname -s)" == "Darwin" ]]; then
+        ARCH=x86_64
+        if [ "$(uname -m)" = "arm64" ]; then
+            ARCH=aarch64
+        fi
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-mac.$ARCH.zip"
+    elif [[ "$(uname -r)" =~ "WSL2" ]]; then
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-win.zip"
+    else
+        archive="${TMPDIR:-/tmp}/Rancher Desktop-linux.zip"
+    fi
+    download_artifact "$archive"
+
+    # Artifacts are zipped, so extract inner ZIP file from outer wrapper.
+    # The outer ZIP has a predictable name like "Rancher Desktop-mac.x86_64.zip"
+    # but the inner one has a version string: "Rancher Desktop-1.7.0-1061-g91ab3831-mac.zip"
+    # Run unzip in "zipinfo" mode, which can print just the file name.
+    local ZIP
+    ZIP="$(unzip -Z -1 "$archive" | head -n1)"
+    if [ -z "$ZIP" ]; then
+        echo "Cannot find inner archive in $(basename "$archive")"
+        exit 1
+    fi
+    local zip_full="${TMPDIR:-/tmp}/$ZIP"
+
+    unzip -o "$archive" "$ZIP" -d "${TMPDIR:-/tmp}"
+    local DEST
+
+    if [[ "$(uname -s)" == "Darwin" ]]; then
+        # Extract from inner archive into /Applications
+        DEST="/Applications"
+        if [ "$RD_LOCATION" = "user" ]; then
+            DEST="$HOME/$DEST"
+        fi
+
+        APP="Rancher Desktop.app"
+        rm -rf "${DEST:?}/$APP"
```

Comment:
Why do we need `${DEST:?}`? Since `$DEST` is always defined here, shouldn't this be just `$DEST`?

And the script is running with `set -o nounset`, so this would already catch this issue, wouldn't it?

Reply:
This silences https://www.shellcheck.net/wiki/SC2115 (otherwise we'd need a shellcheck disable comment). Yes, it's always defined, but shellcheck isn't smart enough to know that.